Invalid alerting params causes panic in Cloudwatch datasources

Grafana will accept the alert to be configured like this ` query(A,1m,now-2m)` but will crash when the query is executed. 

```dashboard.json

{
  "alert": {
    "conditions": [
      {
        "evaluator": {
          "params": [
            1
          ],
          "type": "lt"
        },
        "operator": {
          "type": "and"
        },
        "query": {
          "params": [
            "A",
            "1m",
            "now-2m"
          ]
        },
        "reducer": {
          "params": [],
          "type": "avg"
        },
        "type": "query"
      }
    ],
    "executionErrorState": "keep_state",
    "frequency": "5m",
    "handler": 1,
    "message": "Login test has failed!",
    "name": "Login Test on PROD",
    "noDataState": "alerting",
    "notifications": [
      {
        "id": 1
      }
    ]
  },
  "aliasColors": {},
  "bars": false,
  "dashLength": 10,
  "dashes": false,
  "datasource": "***",
  "description": "Number of successful login attempts from 3 tries.",
  "fill": 1,
  "gridPos": {
    "h": 7,
    "w": 24,
    "x": 0,
    "y": 1
  },
  "id": 2,
  "legend": {
    "alignAsTable": false,
    "avg": false,
    "current": false,
    "hideEmpty": false,
    "hideZero": false,
    "max": false,
    "min": false,
    "rightSide": false,
    "show": false,
    "total": false,
    "values": false
  },
  "lines": true,
  "linewidth": 1,
  "links": [],
  "nullPointMode": "null",
  "percentage": false,
  "pointradius": 5,
  "points": false,
  "renderer": "flot",
  "seriesOverrides": [],
  "spaceLength": 10,
  "stack": false,
  "steppedLine": false,
  "targets": [
    {
      "alias": "Successful login attempts",
      "dimensions": {
        "ServiceName": "***"
      },
      "metricName": "***",
      "namespace": "***",
      "period": "5m",
      "refId": "A",
      "region": "eu-central-1",
      "statistics": [
        "Average"
      ],
      "highResolution": false
    }
  ],
  "thresholds": [
    {
      "colorMode": "critical",
      "fill": true,
      "line": true,
      "op": "lt",
      "value": 1
    }
  ],
  "timeFrom": null,
  "timeShift": null,
  "title": "Login Test on PROD",
  "tooltip": {
    "shared": true,
    "sort": 0,
    "value_type": "individual"
  },
  "type": "graph",
  "xaxis": {
    "buckets": null,
    "mode": "time",
    "name": null,
    "show": true,
    "values": []
  },
  "yaxes": [
    {
      "decimals": 0,
      "format": "short",
      "label": null,
      "logBase": 1,
      "max": null,
      "min": "0",
      "show": true
    },
    {
      "decimals": 0,
      "format": "short",
      "label": null,
      "logBase": 1,
      "max": null,
      "min": "0",
      "show": true
    }
  ],
  "yaxis": {
    "align": false,
    "alignLevel": null
  }
}
```

```bash

t=2018-06-07T09:44:15+0000 lvl=info msg="Alert Rule returned no data" logger=alerting.evalContext ruleId=365 name="Response times of *** endpoints alert" changing state to=no_data
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd711d2]
goroutine 6981 [running]:
github.com/grafana/grafana/pkg/tsdb/cloudwatch.parseResponse(0x0, 0xc4209ffd60, 0x1ffb6e0, 0xbebe5d5f0d5ffb66, 0xfffffff59d9d7102)
/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/cloudwatch.go:348 +0x332
github.com/grafana/grafana/pkg/tsdb/cloudwatch.(*CloudWatchExecutor).executeQuery(0xc42075f460, 0x15182c0, 0xc42155d1a0, 0xc4219222f0, 0xc42075f420, 0x80dcc5, 0x1, 0x0)
/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/cloudwatch.go:191 +0x9aa
github.com/grafana/grafana/pkg/tsdb/cloudwatch.(*CloudWatchExecutor).executeTimeSeriesQuery.func1(0xc42075f460, 0x15182c0, 0xc42155d1a0, 0xc42075f420, 0xc4206e6370, 0xc420da1500, 0xc420da1560, 0x13cba23, 0x1, 0x0)
/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/cloudwatch.go:101 +0x76
created by github.com/grafana/grafana/pkg/tsdb/cloudwatch.(*CloudWatchExecutor).executeTimeSeriesQuery
/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/cloudwatch.go:100 +0x269
```

--- a/pkg/tsdb/cloudwatch/cloudwatch.go
+++ b/pkg/tsdb/cloudwatch/cloudwatch.go
@@ -3,6 +3,7 @@ package cloudwatch
 import (
 	"context"
 	"errors"
+	"fmt"
 	"regexp"
 	"sort"
 	"strconv"
@@ -142,6 +143,10 @@ func (e *CloudWatchExecutor) executeQuery(ctx context.Context, parameters *simpl
 	endTime, err := queryContext.TimeRange.ParseTo()
 	if err != nil {
 		return nil, err
+	}
+
+	if endTime.Before(startTime) {
+		return nil, fmt.Errorf("Invalid time range: End time can't be before start time")
 	}
 
 	params := &cloudwatch.GetMetricStatisticsInput{

